apple/RNSVGRenderable.mm

```diff
@@ -348,33 +352,35 @@ - (void)renderTo:(CGContextRef)context rect:(CGRect)rect
     CGImageRelease(maskImage);
 #else // [macOS
       // Render content of current SVG Renderable to image
-    UIGraphicsBeginImageContextWithOptions(maskBounds.size, NO, 0.0);
+    UIGraphicsBeginImageContextWithOptions(scaledRect.size, NO, 1.0);
     CGContextRef newContext = UIGraphicsGetCurrentContext();
-    CGContextTranslateCTM(newContext, 0.0, height);
-    CGContextScaleCTM(newContext, 1.0, -1.0);
-    [self renderLayerTo:newContext rect:rect];
+    CGContextConcatCTM(newContext, CGAffineTransformInvert(CGContextGetCTM(newContext)));
+    CGContextConcatCTM(newContext, screenScaleCTM);
+    CGContextConcatCTM(newContext, currentCTM);
+    [self renderLayerTo:newContext rect:scaledRect];
     CGImageRef contentImage = CGBitmapContextCreateImage(newContext);
     UIGraphicsEndImageContext();
 
     // Blend current element and mask
-    UIGraphicsBeginImageContextWithOptions(maskBounds.size, NO, 0.0);
+    UIGraphicsBeginImageContextWithOptions(scaledRect.size, NO, 1.0);
     newContext = UIGraphicsGetCurrentContext();
-    CGContextTranslateCTM(newContext, 0.0, height);
-    CGContextScaleCTM(newContext, 1.0, -1.0);
+    CGContextConcatCTM(newContext, CGAffineTransformInvert(CGContextGetCTM(newContext)));
 
     CGContextSetBlendMode(newContext, kCGBlendModeCopy);
-    CGContextDrawImage(newContext, maskBounds, maskImage);
+    CGContextDrawImage(newContext, scaledRect, maskImage);
     CGImageRelease(maskImage);
 
     CGContextSetBlendMode(newContext, kCGBlendModeSourceIn);
-    CGContextDrawImage(newContext, maskBounds, contentImage);
+    CGContextDrawImage(newContext, scaledRect, contentImage);
     CGImageRelease(contentImage);
 
     CGImageRef blendedImage = CGBitmapContextCreateImage(newContext);
     UIGraphicsEndImageContext();
 
     // Render blended result into current render context
-    CGContextDrawImage(context, maskBounds, blendedImage);
+    CGContextConcatCTM(context, CGAffineTransformInvert(currentCTM));
+    CGContextDrawImage(context, rect, blendedImage);
+    CGContextConcatCTM(context, currentCTM);
```

Comment:
Why do we have to concat at the end?

Reply:
Since it's the "original" `context`, we need to restore it to its original state so that the subsequent elements render correctly.